src/utils/path.js

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import type { CallNodePath } from '../types/profile-derived';
+
+export function arePathsEqual(a: CallNodePath, b: CallNodePath): boolean {
+  if (a === b) {
+    return true;
+  }
+
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  // Iterating from the end because this will likely fail faster
+  for (let i = a.length - 1; i >= 0; i--) {
+    if (a[i] !== b[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// We take the easy path by converting the path to a string that will be unique.
+// In the future and if necessary we might want to have an integer-based hash so
+// that it's faster. In that case we might want to use only the n last values
+// that should be more discriminate and keep it fast.
```

Comment:
This hash function has a nice property: it's a perfect hash function. So we don't need to allocate buckets and checks for equality all the time. I tried this in another PR and the performance was less good, but we can revisit in a future patch.
(I'll add something about it as a comment but can't do it now)